### PR TITLE
Add API tests for complete order and cancel order actions

### PR DESCRIPTION
**Pull request by uLucasFraga**
### Description 

I'm creating new test scenarios to **cancel and delete an order** in the cart.

The task can be [seen here](https://github.com/stefanteixeira/demo-playwright-test/projects/1#card-83560698).

### How to test

- `yarn install && yarn test:api`

### Checklist

- [x] Branch name follows the pattern: `concise-feature-description`
- [x] PR has a descriptive name
- [x] PR branch is up-to-date with `main` (if not, rebase it)
- [x] Double-check the quality of [commit messages](http://chris.beams.io/posts/git-commit/)

### CODE REVIEW

File: tests/api/cart.api.test.js [hunk 1]
```diff
@@ -1,41 +1,120 @@
...
-    
+
+    await expect(response).toBeOK()
+  })
+
+  test('retrieves an existing cart by its id', async ({ request }) => {
+    const idProduct = await getProductId(request, authorization, getProductBody())
+    const _id = await getCartId(request, authorization, getCartBody(idProduct))
+
+    const response = await request.get(`/carrinhos/${_id}`)
```
Comment (stefanteixeira) [inline tests/api/cart.api.test.js:22]:
```suggestion
    const productId = await getProductId(request, authorization, getProductBody())
    const cartId = await getCartId(request, authorization, getCartBody(productId))

    const response = await request.get(`/carrinhos/${cartId}`)
```

File: tests/api/cart.api.test.js [hunk 2]
```diff
@@ -1,41 +1,120 @@
...
+
     await expect(response).toBeOK()
   })
 
   test('creates a cart successfully', async ({ request }) => {
-    const id = await getProductId(request, authorization, getProductBody())
+    const authorization = await getAuthToken(request)
+    const idProduct = await getProductId(request, authorization, getProductBody())
```
Comment (stefanteixeira) [inline tests/api/cart.api.test.js:29]:
```suggestion
    const productId = await getProductId(request, authorization, getProductBody())
```

File: tests/api/cart.api.test.js [hunk 3]
```diff
@@ -1,41 +1,120 @@
...
 
     const response = await request.post('/carrinhos', {
-      data: getCartBody(id),
+      data: getCartBody(idProduct),
```
Comment (stefanteixeira) [inline tests/api/cart.api.test.js:32]:
```suggestion
      data: getCartBody(productId),
```

File: tests/api/cart.api.test.js [hunk 4]
```diff
@@ -1,41 +1,120 @@
...
       headers: { 'Authorization': authorization }
     })
 
     await expect(response).toBeOK()
 
     const responseBody = JSON.parse(await response.text())
-    
+
     await expect(responseBody.message).toBe('Cadastro realizado com sucesso')
```
Comment (stefanteixeira) [inline tests/api/cart.api.test.js:40]: @uLucasFraga let's not hardcode messages, we don't have access to the codebase. Ideally we would import constants from these messages and use in tests, but it's not the case 😄 
According to the API specification, the only OK scenario would be that one, so it won't be necessary to validate the message.

File: tests/api/cart.api.test.js [hunk 5]
```diff
@@ -1,41 +1,120 @@
...
   })
 
-  test('retrieves an existing cart by its id', async ({ request }) => {
+  test('delete an order successfully', async ({ request }) => {
```
Comment (stefanteixeira) [inline tests/api/cart.api.test.js:43]:
```suggestion
  test('deletes an order successfully', async ({ request }) => {
```

File: tests/api/cart.api.test.js [hunk 6]
```diff
@@ -1,41 +1,120 @@
...
     const idProduct = await getProductId(request, authorization, getProductBody())
-    const _id = await getCartId(request, authorization, getCartBody(idProduct))
-    
-    const response = await request.get(`/carrinhos/${_id}`)
-    
+    await getCartId(request, authorization, getCartBody(idProduct))
```
Comment (stefanteixeira) [inline tests/api/cart.api.test.js:45]: @uLucasFraga let's change all occurrences of `idProduct` to `productId`, please

File: tests/api/cart.api.test.js [hunk 7]
```diff
@@ -1,41 +1,120 @@
...
+
+    const response = await request.delete('/carrinhos/concluir-compra', {
+      headers: { 'Authorization': authorization }
+    })
+
+    await expect(response).toBeOK()
+
+    const responseBody = JSON.parse(await response.text())
+
+    await expect(responseBody.message).toBe('Registro excluído com sucesso')
```
Comment (stefanteixeira) [inline tests/api/cart.api.test.js:55]: @uLucasFraga same as the [previous comment](https://github.com/stefanteixeira/demo-playwright-test/pull/34/files#r937202926), no need to have the check for a hardcoded text

Comment (stefanteixeira) [inline tests/api/cart.api.test.js:55]: @uLucasFraga same as the [previous comment](https://github.com/stefanteixeira/demo-playwright-test/pull/34/files#r937202926)

File: tests/api/cart.api.test.js [hunk 8]
```diff
@@ -1,41 +1,120 @@
...
+  })
+
+  test('delete an order without a cart', async ({ request }) => {
```
Comment (stefanteixeira) [inline tests/api/cart.api.test.js:58]:
```suggestion
  test('deletes an order without a cart', async ({ request }) => {
```

File: tests/api/cart.api.test.js [hunk 9]
```diff
@@ -1,41 +1,120 @@
...
+    const response = await request.delete('/carrinhos/concluir-compra', {
+      headers: { 'Authorization': authorization }
+    })
+
+    await expect(response).toBeOK()
+
+    const responseBody = JSON.parse(await response.text())
+
+    await expect(responseBody.message).toBe('Não foi encontrado carrinho para esse usuário')
```
Comment (stefanteixeira) [inline tests/api/cart.api.test.js:67]: @uLucasFraga same as the [previous comment](https://github.com/stefanteixeira/demo-playwright-test/pull/34/files#r937202926)

File: tests/api/cart.api.test.js [hunk 10]
```diff
@@ -1,41 +1,120 @@
...
+  })
+
+  test('delete an order without a token', async ({ request }) => {
```
Comment (stefanteixeira) [inline tests/api/cart.api.test.js:70]:
```suggestion
  test('fails to delete an order if authorization token is not provided', async ({ request }) => {
```

File: tests/api/cart.api.test.js [hunk 11]
```diff
@@ -1,41 +1,120 @@
...
+    })
+
+    await expect(response).not.toBeOK()
+
+    const responseBody = JSON.parse(await response.text())
+
+    await expect(responseBody.message).toBe('Token de acesso ausente, inválido, expirado ou usuário do token não existe mais')
+  })
+
+  test('cancel an order successfully', async ({ request }) => {
```
Comment (stefanteixeira) [inline tests/api/cart.api.test.js:82]:
```suggestion
  test('cancels an order successfully', async ({ request }) => {
```

File: tests/api/cart.api.test.js [hunk 12]
```diff
@@ -1,41 +1,120 @@
...
+    })
+
+    await expect(response).toBeOK()
+
+    const responseBody = JSON.parse(await response.text())
+
+    await expect(responseBody.message).toBe('Registro excluído com sucesso. Estoque dos produtos reabastecido')
+  })
+
+  test('cancel an order without a cart', async ({ request }) => {
```
Comment (stefanteixeira) [inline tests/api/cart.api.test.js:97]:
```suggestion
  test('cancels an order without a cart', async ({ request }) => {
```

File: tests/api/cart.api.test.js [hunk 13]
```diff
@@ -1,41 +1,120 @@
...
+    })
+
     await expect(response).toBeOK()
+
+    const responseBody = JSON.parse(await response.text())
+
+    await expect(responseBody.message).toBe('Não foi encontrado carrinho para esse usuário')
+  })
+
+  test('cancel an order without a token', async ({ request }) => {
```
Comment (stefanteixeira) [inline tests/api/cart.api.test.js:109]:
```suggestion
  test('fails to cancel an order if authorization token is not provided', async ({ request }) => {
```

File: tests/api/cart.api.test.js [hunk 14]
```diff
@@ -32,12 +33,10 @@ test.describe.parallel('Carts API', () => {
...
       headers: { 'Authorization': authorization }
     })
-
-    await expect(response).toBeOK()
-
     const responseBody = JSON.parse(await response.text())
 
-    await expect(responseBody.message).toBe('Cadastro realizado com sucesso')
+    await expect(response).toBeOK()
+    expect(responseBody.message).toEqual(POST_SUCESS)
```
Comment (stefanteixeira) [inline tests/api/cart.api.test.js:39]: @uLucasFraga let's not hardcode any text message, we should get these from the codebase instead. I managed to get the constants from serverest repository, please follow this example and apply to other tests from the PR 👍 

Please fix the following things in your test as well:

- `expect(responseBody.message)` is not async, so it doesn't need an `await`
- Let's order the expects like shown in this test
- Use `.toEqual` to expect for any primitive, `.toBe` works but it's more suited for object comparison
- Refer to [serverest constants file](https://github.com/ServeRest/ServeRest/blob/trunk/src/utils/constants.js) to identify which ones you should import and use

File: tests/api/cart.api.test.js (hunk 6 above)
Comment (uLucasFraga) [reply tests/api/cart.api.test.js:44]: sure!

File: tests/api/cart.api.test.js (hunk 14 above)
Comment (uLucasFraga) [reply tests/api/cart.api.test.js:39]: oh gosh, that's amazing!
I'll try now. 🛸

**Pull request closed without merging**